sound/soc/sof/intel/hda.c

```diff
@@ -380,7 +380,8 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 	/* create codec instances */
 	hda_codec_probe_bus(sdev, hda_codec_use_common_hdmi);
 
-	hda_codec_i915_put(sdev);
+	if (!HDA_IDISP_CODEC(bus->codec_mask))
+		hda_codec_i915_put(sdev);
```

Comment:
@kv2019i have you tried some suspend/resume cycles with this change? Wont this cause an imbalance in the HDMI codec ref counts?

Reply:
@ranj063 Actually this is the reason I added the get/put refactoring patch to the series. This change looks too suspicious otherwise. :) Now I reordered the series so the refactoring is first and then this change looks more sane. But yes, I did test suspend/resume cycles and no issues found. The hda setup is a bit complex as SOF hda code does stage1, hdac_hda the 2nd and then the actual codec driver does the rest. We basicly keep the display power on for stages 1&2 and let the codec driver handle runtime power management. Powering down between 1&2 is not useful.

Reply:
@kv2019i thanks for the clarification!